core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/time/TimeFormat.java

```diff
@@ -14,31 +14,45 @@
  */
 public class TimeFormat implements DataInstruction {
 
-    private final String ORIGINAL_TIME_REPRESENTATION_KEY = "OriginalDateRepresentation";
+    private final static String ORIGINAL_TIME_REPRESENTATION = "OriginalTimeRepresentation";
 
-    private final String DESIRED_TIME_REPRESENTATION_KEY = "DesiredDateRepresentation";
+    private final static String ORIGINAL_TIME_REPRESENTATION_FORMAT = "OriginalTimeRepresentationFormat";
 
-    private final Pattern INPUT_PARAMETER_PATTERN = Pattern.compile(
-            "\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION_KEY + ">\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}.\\d{3})\"?\\s*,\\s*\"?(?<"
-                    + DESIRED_TIME_REPRESENTATION_KEY + ">[^\"]+)\"?\\s*");
+    private final static String DESIRED_TIME_REPRESENTATION = "DesiredTimeRepresentation";
 
-    private final SimpleDateFormat ORIGINAL_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+    private final static Pattern INPUT_PARAMETERS_PATTERN = Pattern.compile(
+            "\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION + ">[^\"]+)\"?\\s*" +
+                    "(,\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION_FORMAT + ">[^\"]+)\"?\\s*)?"+
+                    ",\\s*\"?(?<" +  DESIRED_TIME_REPRESENTATION + ">[^\"]+)\"?");
+
+
+    //date format by default
+    private static SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
 
     @Override
     public String generateOutput(String parameters) {
-        Matcher inputParameterMatcher = INPUT_PARAMETER_PATTERN.matcher(parameters);
+        Matcher inputParameterMatcher = INPUT_PARAMETERS_PATTERN.matcher(parameters);
+
         if (!inputParameterMatcher.find()) {
             throw new IllegalArgumentException(MessageFormat.format("Illegal arguments provided to " + this.getKeyword() + ": {0}", parameters));
         } else {
-            try {
-                Date originalDate = ORIGINAL_DATE_FORMAT.parse(inputParameterMatcher.group(ORIGINAL_TIME_REPRESENTATION_KEY));
-                SimpleDateFormat desiredDateRepresentation = new SimpleDateFormat(
-                        inputParameterMatcher.group(DESIRED_TIME_REPRESENTATION_KEY));
-                return desiredDateRepresentation.format(originalDate);
-            } catch (ParseException e) {
-                throw new IllegalArgumentException(MessageFormat.format("Cannot generate Time from {0}",
-                        inputParameterMatcher.group(ORIGINAL_TIME_REPRESENTATION_KEY)));
+            if (inputParameterMatcher.group(ORIGINAL_TIME_REPRESENTATION_FORMAT) != null) {
```

Comment:
isn't this always non-null as it is the second parameter?

Reply:
nop, it depends on if the user introduced 2 or 3 arguments, at this line I check if the user inserted a custom time format

Reply:
so, 3 arguments in this case

Reply:
the arguments : ORIGINAL_TIME_REPRESENTATION_FORMAT is "optional"=> **(,**\\s*\"?(?<" + ORIGINAL_TIME_REPRESENTATION_FORMAT + ">[^\"]+)\"?\\s***)?**